From c064c801c1e4f6a710487ece640825e3c63cb030 Mon Sep 17 00:00:00 2001
From: Sai Surya Teja <sssteja@habana.ai>
Date: Wed, 09 Feb 2022 12:57:22 +0200
Subject: [PATCH] [SW-62889] gaudi2: sec: decouple dcore0_rtr0 lbw rr in pcie rr

Current logic for PCI LBW RR configures even dcore0_rtr0 as this
rtr blocks pcie inbound traffic as well. But, this rtr also blocks
other dcore engines like PCIE_VDEC, PMMU, etc as well.

So, it is decided not to configure dcore0_rtr0 for PCIE LBW RR
range as the inbound PCIE traffic is anyways controlled by PCIE
MSTR RR.

Hence, this patch removes the dcore0_rtr0 from pcie rr config.

Change-Id: I6d340c7b86c26ae69d153e12c3afe5a38d7b08a2
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index a19fc4c..0e49d94 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -308,9 +308,6 @@
 
 	if (rr_sel.pcie) {
 		hl_secure_lbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PCIE_INST_NUM, HL_SEC_ENGINE_PCIE);
-		if (!rr_sel.noc)
-			hl_secure_lbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PCIE_RTR_NUM,
-						HL_SEC_ENGINE_IF_RTR);
 	}
 
 	if (rr_sel.psoc)
